mastrms/mastrms/api/dump.py

```diff
@@ -0,0 +1,138 @@
+from functools import wraps
+from django.db.models import Q
+from rest_framework import serializers, viewsets, permissions, fields
+from rest_framework.renderers import JSONRenderer
+from rest_framework.request import Request
+
+
+from .base import router
+from .repository import ProjectSerializer, InvestigationSerializer
+from .repository import get_user_projects
+from .experiment import ExperimentSerializer, ExperimentStatusSerializer
+from .experiment import UserInvolvementTypeSerializer, InstrumentMethodSerializer
+from .experiment import UserExperimentSerializer
+from .experiment import ClientFileSerializer
+from .experiment import RuleGeneratorSerializer, NodeClientSerializer
+from .sample import SampleSerializer, RunSampleSerializer, RunSerializer
+from .sample import ComponentGroupSerializer, ComponentSerializer
+from .users import UserSerializer
+from ..repository.models import Project
+from ..repository.isatab import ISATabExportView
+from ..repository.models import ExperimentStatus, UserInvolvementType
+from ..repository.models import UserExperiment, InstrumentMethod
+from ..repository.models import Run, RunSample, Sample
+from ..repository.models import NodeClient, RuleGenerator
+from ..repository.models import Component, ComponentGroup
+from ..repository.models import ClientFile
+from ..users.models import User
+
+
+def cache_related(f):
+    @wraps(f)
+    def wrapped(self, *args, **kwargs):
+        self.load_related()
+        return f(self, *args, **kwargs)
+    return wrapped
+
+def cached_prop(name):
+    load_get = lambda self: getattr(self.do_load_related(), name)
+    return property(cache_related(load_get))
+
+class DumpProject(Project):
```

Comment:
Any reason to load all the properties at once in do_load_related?
If not to my eyes it would look more straightforward if each property was a method that gets what it needs and it is cached.
Something like in: https://pypi.python.org/pypi/cached-property/1.2.0 

@cached_property
def experiments(self):
    return self.experiment_set.all()

@cached_property
def samples(self):
    return Sample.objects.filter(experiment__project=self)

@cached_property
def user_experiment(self):
    return UserExperiment.objects.filter(experiment__in=self.experiments)

etc.